Makefile

```diff
@@ -41,41 +64,72 @@ bumposs: toolchain
 	GOPROXY=direct go get tailscale.com@main
 	go mod tidy -compat=1.20
 
-toolchain:
-ifneq ($(TOOLCHAINWANT),$(TOOLCHAINSUM))
-	@echo want: $(TOOLCHAINWANT)
-	@echo got: $(TOOLCHAINSUM)
+$(TOOLCHAINDIR)/go/bin/go:
```

Comment:
If I understand this correctly, removing the ifneq TOOLCHAINWANT,TOOLCHAINSUM means it will remove ${HOME}/.cache/tailscale-android-go-* and re-download it on every build?

One problem we've had is that GitHub expires the build artifacts more rapidly than we update the Android toolchain checksum. I'm concerned that if we expect to re-download each time, it will only work for a couple weeks and then Android builds begin failing.

Reply:
It won't download it each time - currently our android toolchain builds contain version information in the path, i.e. `~/.cache/tailscale-android-go-40dc4d834a5fde9872bcf470be50069f56c3e3b3/` so this will only fetch the go toolchain if that path does not exist.

If we need to override versions in the future we can produce a stamp file that is a prerequisite and have the conditional update the stamp, which would retrigger the action.

Reply:
I think we need to address the issue you describe here with our expiring toolchains separately. I'll make a bug for that.

Reply:
tailscale/corp#13478